gremlin-go/driver/client.go

```diff
@@ -71,19 +78,64 @@ func NewClient(url string, configurations ...func(settings *ClientSettings)) (*C
 		transporterType: settings.TransporterType,
 		connection:      conn,
 	}
+	// TODO: PoolSize must be 1 on session mode. Implement after AN-980
 	return client, nil
 }
 
 // Close closes the client via connection.
 func (client *Client) Close() error {
+	if client.IsClosed() {
+		return nil
+	}
+	// If it is a Session, call closeSession
+	if client.session != "" {
+		_, err := client.closeSession()
+		if err != nil {
+			return err
+		}
+	}
+	client.logHandler.logger.Logf(Info, "Closing Client with url %s", client.url)
 	return client.connection.close()
 }
 
+func (client *Client) IsClosed() bool {
+	return client.closed
+}
+
 // Submit submits a Gremlin script to the server and returns a ResultSet.
-func (client *Client) Submit(traversalString string) (ResultSet, error) {
+func (client *Client) Submit(message interface{}) (ResultSet, error) {
 	// TODO AN-982: Obtain connection from pool of connections held by the client.
-	client.logHandler.logf(Debug, submitStartedString, traversalString)
-	request := makeStringRequest(traversalString, client.traversalSource)
+	client.logHandler.logf(Debug, submitStarted, message)
+	args := map[string]interface{}{
+		"gremlin": message,
+		"aliases": map[string]interface{}{
+			"g": client.traversalSource,
+		},
+	}
+	var processor string
+	var op string
+	switch message.(type) {
+	case bytecode:
```

Comment:
If we are adding a `bytecode` type check here, is `submitBytecode` still needed?

Reply:
Let's keep submitBytecode, we shouldn't allow a user to submit bytecode directly.

Reply:
It's an internal type.